From 5cf079aa731c75a2a7ab2ada966f76fb477acb18 Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Tue, 11 Apr 2023 22:52:54 +0530
Subject: [PATCH] Fix for green overlay during video playback

Gralloc buffer has been aligned with 64 lines whereas the surface is
created with 32 alignment resulting in green overlay.

Fix the issue by modifying the alignment to 64 while calculating
offsets.

Change-Id: I89346107bfa5658230ebb42846dc0651fba31aa4
Tracked-On: OAM-108761
Signed-off-by: Marc Mao <marc.mao@intel.com>
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 c2_utils/src/mfx_va_allocator.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/c2_utils/src/mfx_va_allocator.cpp b/c2_utils/src/mfx_va_allocator.cpp
index f7ad584..7f48b68 100755
--- a/c2_utils/src/mfx_va_allocator.cpp
+++ b/c2_utils/src/mfx_va_allocator.cpp
@@ -538,7 +538,7 @@ mfxStatus MfxVaFrameAllocator::CreateSurfaceFromGralloc(const MfxGrallocModule::
     surfExtBuf.pitches[3] = 0;
     surfExtBuf.offsets[0] = 0;
     // Gralloc buffer has been aligned with 32 pixels for decoder
-    surfExtBuf.offsets[1] = decode_target ? info.pitches[0] * ((height + 31) & ~31) : info.pitches[0] * ((height + 15) & ~15);
+    surfExtBuf.offsets[1] = decode_target ? info.pitches[0] * ((height + 63) & ~63) : info.pitches[0] * ((height + 15) & ~15);
     surfExtBuf.offsets[2] = 0;
     surfExtBuf.offsets[3] = 0;
     surfExtBuf.data_size = decode_target ? info.pitches[0] * ((height + 31) & ~31) * 1.5 : info.pitches[0] * ((height + 15) & ~15) * 1.5;
-- 
2.40.0

